### PR TITLE
Add RISC-V Support

### PR DESCRIPTION
**Pull request by bradjc**
Updates crt0.c and tock.c to support RISC-V.

Also adds newlib support for riscv.

### CODE REVIEW

File: libtock/tock.c [hunk 1]
```diff
@@ -40,6 +40,12 @@ void yield_for(bool *cond) {
   }
 }
 
+#if defined(__thumb__)
+
+//
```
Comment (ppannuto) [inline libtock/tock.c]:
```suggestion
```

File: libtock/tock.c [hunk 2]
```diff
@@ -40,6 +40,12 @@ void yield_for(bool *cond) {
...
+// IMPLEMENATTION FOR CORTEX-M THUMB BASED PLATFORMS
```
Comment (ppannuto) [inline libtock/tock.c]:
```suggestion
```

Cosmetic / consistency thing: I think these comments are redundant with the `if defined` clause above -- if you really think they're necessary then they should probably be added to the crt0 as well

edit: the lack of a multi-line suggestion in github is criminal

File: libtock/tock.c [hunk 3]
```diff
@@ -40,6 +40,12 @@ void yield_for(bool *cond) {
...
+//
```
Comment (ppannuto) [inline libtock/tock.c]:
```suggestion
```

File: libtock/tock.c [hunk 4]
```diff
@@ -136,6 +142,91 @@ void* memop(uint32_t op_type, int arg1) {
   return ret;
 }
 
+#elif defined(__riscv)
+
+//
```
Comment (ppannuto) [inline libtock/tock.c]:
```suggestion
```

File: libtock/tock.c [hunk 5]
```diff
@@ -136,6 +142,91 @@ void* memop(uint32_t op_type, int arg1) {
...
+// IMPLEMENATTION FOR RISC-V BASED PLATFORMS
```
Comment (ppannuto) [inline libtock/tock.c]:
```suggestion
```

File: libtock/tock.c [hunk 6]
```diff
@@ -136,6 +142,91 @@ void* memop(uint32_t op_type, int arg1) {
...
+//
```
Comment (ppannuto) [inline libtock/tock.c]:
```suggestion
```

File: libtock/tock.c (hunk 2 above)
Comment (niklasad1) [reply libtock/tock.c]: Can you fix the typo in `implementation` too? :P

File: libtock/tock.c (hunk 5 above)
Comment (niklasad1) [reply libtock/tock.c]: s/IMPLEMENATTION/IMPLEMENTATION

**Pull request merged**